provider/cluster/kube/builder/workload.go

```diff
@@ -0,0 +1,180 @@
+package builder
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/tendermint/tendermint/libs/log"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	manifesttypes "github.com/ovrclk/akash/manifest"
+	clusterUtil "github.com/ovrclk/akash/provider/cluster/util"
+	mtypes "github.com/ovrclk/akash/x/market/types"
+)
+
+const (
+	StorageAttributePersistent = "persistent"
+	StorageAttributeClass      = "class"
+	StorageAttributeMount      = "mount"
+	StorageAttributeReadOnly   = "readOnly"
+)
+
+type workloadBase interface {
+	builderBase
+	Name() string
+}
+
+type workload struct {
+	builder
+	service          *manifesttypes.Service
+	runtimeClassName string
+}
+
+var _ workloadBase = (*workload)(nil)
+
+func newWorkloadBuilder(log log.Logger, settings Settings, lid mtypes.LeaseID, group *manifesttypes.Group, service *manifesttypes.Service) workload {
+	return workload{
+		builder: builder{
+			settings: settings,
+			log:      log.With("module", "kube-builder"),
+			lid:      lid,
+			group:    group,
+		},
+		service:          service,
+		runtimeClassName: settings.DeploymentRuntimeClass,
+	}
+}
+
+func (b *workload) container() corev1.Container {
+	falseValue := false
+
+	kcontainer := corev1.Container{
+		Name:    b.service.Name,
+		Image:   b.service.Image,
+		Command: b.service.Command,
+		Args:    b.service.Args,
+		Resources: corev1.ResourceRequirements{
+			Limits:   make(corev1.ResourceList),
+			Requests: make(corev1.ResourceList),
+		},
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		SecurityContext: &corev1.SecurityContext{
+			RunAsNonRoot:             &falseValue,
+			Privileged:               &falseValue,
+			AllowPrivilegeEscalation: &falseValue,
+		},
+	}
+
+	if cpu := b.service.Resources.CPU; cpu != nil {
+		requestedCPU := clusterUtil.ComputeCommittedResources(b.settings.CPUCommitLevel, cpu.Units)
+		kcontainer.Resources.Requests[corev1.ResourceCPU] = resource.NewScaledQuantity(int64(requestedCPU.Value()), resource.Milli).DeepCopy()
+		kcontainer.Resources.Limits[corev1.ResourceCPU] = resource.NewScaledQuantity(int64(cpu.Units.Value()), resource.Milli).DeepCopy()
+	}
+
+	if mem := b.service.Resources.Memory; mem != nil {
+		requestedMem := clusterUtil.ComputeCommittedResources(b.settings.MemoryCommitLevel, mem.Quantity)
+		kcontainer.Resources.Requests[corev1.ResourceMemory] = resource.NewQuantity(int64(requestedMem.Value()), resource.DecimalSI).DeepCopy()
+		kcontainer.Resources.Limits[corev1.ResourceMemory] = resource.NewQuantity(int64(mem.Quantity.Value()), resource.DecimalSI).DeepCopy()
+	}
+
+	for _, storage := range b.service.Resources.Storage {
+		attr := storage.Attributes.Find(StorageAttributeMount)
+		if mount, _ := attr.AsString(); mount != "" {
+			kcontainer.VolumeMounts = append(kcontainer.VolumeMounts, corev1.VolumeMount{
+				// matches VolumeName in persistentVolumeClaims below
+				Name: fmt.Sprintf("%s-%s", b.service.Name, storage),
```

Comment:
What does this name look like? It's calling `%s` on an object, which I know works but im trying to figure out what is actually created here

Reply:
There was a missing field.
It is now `Name: fmt.Sprintf("%s-%s", b.service.Name, storage.Name),`